src/qt/test/trafficgraphdatatests.cpp

```diff
@@ -0,0 +1,172 @@
+#include "trafficgraphdatatests.h"
+#include "../trafficgraphdata.h"
+#include <algorithm>
+#include <sstream>
+#include <QTime>
+
+void TrafficGraphDataTests::simpleCurrentSampleQueueTests()
+{
+    TrafficGraphData trafficGraphData(TrafficGraphData::Range_5m);
+    for(int i=0; i<TrafficGraphData::DESIRED_DATA_SAMPLES; i++)
+        QVERIFY(trafficGraphData.update(TrafficSample(i,i)));
+
+    TrafficGraphData::SampleQueue queue = trafficGraphData.getCurrentRangeQueue();
+    QCOMPARE(queue.size(), TrafficGraphData::DESIRED_DATA_SAMPLES);
+    for(int i=0; i<TrafficGraphData::DESIRED_DATA_SAMPLES; i++){
+        QCOMPARE((int)queue.at(i).in, TrafficGraphData::DESIRED_DATA_SAMPLES-i-1);
+        QCOMPARE((int)queue.at(i).out, TrafficGraphData::DESIRED_DATA_SAMPLES-i-1);
+    }
+
+    QVERIFY(trafficGraphData.update(TrafficSample(0,0)));
+    queue = trafficGraphData.getCurrentRangeQueue();
+    QCOMPARE(queue.size(), TrafficGraphData::DESIRED_DATA_SAMPLES);
+    QCOMPARE((int)queue.at(0).in, 0);
+    QCOMPARE((int)queue.at(0).out, 0);
+}
+
+namespace {
+
+void checkQueue(const TrafficGraphData::SampleQueue& queue, int value)
+{
+    for(int i=0;i<queue.size();i++){
+        std::ostringstream oss;
+        oss<< "i:" << i << " value:" << value << " actual:" << queue.at(i).in;
+        QVERIFY2(value ==(int)queue.at(i).in, oss.str().c_str());
+        QVERIFY2(value == (int)queue.at(i).out, oss.str().c_str());
+    }
+}
+
+void testQueueFill(TrafficGraphData::GraphRange range, int multiplier)
+{
+    int size = 10000;
+    TrafficGraphData trafficGraphData(range);
+    for(int i=1; i<=size; i++){
+        bool result = trafficGraphData.update(TrafficSample(1,1));
+        std::ostringstream oss;
+        oss<< "result:" << result << " multiplier:" << multiplier << " i:" << i << " range:" << range;
+        if (i==1){
+            if (range == TrafficGraphData::Range_5m)
+                QVERIFY2(result, oss.str().c_str());
+            else
+                QVERIFY2(!result, oss.str().c_str());
+        }
+        else if (i%multiplier == 0)
```

Comment:
code formatting: need curly brackets here and in `else` below since `if` already has them